### PR TITLE
chore(docs): document HTTPRequest with TSDoc

### PR DESCRIPTION
**Pull request by jackfranklin**

### CODE REVIEW

File: src/common/HTTPRequest.ts [hunk 1]
```diff
@@ -66,44 +143,120 @@ export class HTTPRequest {
...
+   * @returns one of the following: `document`, `stylesheet`, `image`, `media`,
+   * `font`, `script`, `texttrack`, `xhr`, `fetch`, `eventsource`, `websocket`,
+   * `manifest`, `other`.
+   */
   resourceType(): string {
+    // TODO (@jackfranklin): protocol.d.ts has a type for this, but all the
+    // string values are uppercase. The Puppeteer docs explicitly say the
+    // potential values are all lower case, and the constructor takes the event
+    // type and calls toLowerCase() on it, so we can't reuse the type from the
+    // protocol.d.ts. Why do we lower case?
```
Comment (jackfranklin) [inline src/common/HTTPRequest.ts:166]: @mathiasbynens I don't suppose you have any insight here?

File: src/common/HTTPRequest.ts [hunk 2]
```diff
@@ -66,44 +143,120 @@ export class HTTPRequest {
...
     return this._redirectChain.slice();
   }
 
   /**
-   * @returns {?{errorText: string}}
+   * Access information about the request's failure.
+   *
+   * @remarks
+   *
+   * Example of logging all failed requests:
```
Comment (hanselfmu) [inline src/common/HTTPRequest.ts:250]: Do we want to separate these in a `@example` tag?

**Review by hanselfmu — APPROVED** [review]
LGTM % one small nit, and it can go either way. Thanks Jack!

File: src/common/HTTPRequest.ts (hunk 2 above)
Comment (jackfranklin) [reply src/common/HTTPRequest.ts:250]: Yes, I think we should! Good catch, thanks :)

File: src/common/HTTPRequest.ts (hunk 1 above)
Comment (mathiasbynens) [reply src/common/HTTPRequest.ts:166]: I suppose for parity with other Puppeteer APIs? We don't really use uppercase string values anywhere.

**Pull request merged**